bloggitt/core/views.py

```diff
@@ -48,9 +48,47 @@ def postlist(request):
     post_list=Post.objects.all().order_by('-created_on')
     return render(request,'index.html',{'post_list':post_list})
 
+
 def postdetail(request, slug):
+    if not request.user.is_authenticated:
+        return redirect('login')
+        
+    post = Post.objects.get(slug=slug)
+    Favourites,_ = FavouritePost.objects.get_or_create(user=request.user)
+    post_in_favorites = None
+    if post in Favourites.posts.all():
+        post_in_favorites = True
+    else:
+        post_in_favorites = False
+
+    return render(request, 'detail.html', {'post': post, 'post_in_favorites': post_in_favorites})
+
+
+def Favorites(request, slug):
+    if not request.user.is_authenticated:
+        return redirect('login')
+
+    user = request.user
+    Favourites,_ = FavouritePost.objects.get_or_create(user=user)
+
     post = Post.objects.get(slug=slug)
-    return render(request, 'detail.html', {'post': post})
+
+    if post not in Favourites.posts.all():
+        Favourites.posts.add(post)
+    else:
+        Favourites.posts.remove(post)
+    
+    Favourites.save()
+    
+    return redirect('post_detail', slug=slug)
```

Comment:
According to me, whenever a user clicks on Add to Favorite button, the post is added to his fav and page reloads and button changes to remove from fav, isn't it?? I don't want page reload everytime, the button should change without page reload and adding or removing from fav should also happen.

Reply:
I have used ajax so as not to reload the page, but also doing the backend work successfully.

Reply:
Okayy @SubhradeepSS , Good work!!